### ISSUE TITLE
sendSyncRequest does not restore contacts

### ISSUE THREAD

**Issue by exquo** [AsamK/signal-cli#622]
Thank you for adding the `sendSyncRequest` command! It should come handy in a few situations.

However something does not seem to work as expected: I have tried deleting one of the groups (or all of them) in the `groupStore` section of signal-cli's data file, and then executing

	signal-cli sendSyncRequest

under the assumption that it would restore the deleted entries (I did make a backup prior :). But the data file remains unchanged.

Deleting one of the contacts in the `recipients-store` file and running `sendSyncRequest` does not restore the deleted recipient either.

Adding the `--verbose` flag produces no output.

Is this command supposed to work the way I expect it to work?

For comparison, I have done `CONTACTS_SYNC` from the master signal-android device, and the v1 groups (but not v2) have been added to the `groupStore`. Is that another issue?
Update: sometime later I have received a `GROUPS_SYNC` message too (but by then I have already restored the `groupStore` from the backup). Anyway, the last "issue" is probably not an issue.

**Comment by AsamK** [comment]
After executing the `sendSyncRequest` command you should receive a complete sync contacts and sync groups message from the master device, though it might take a bit for the app to handle the request. In my tests contacts are restored correctly.
In general recipients shouldn't be deleted completely from the recipients-store, as the id is used to link them to sessions and identities. Deleting the `contact` field from a recipient is fine.
v2 groups are not included in the sync message, for those signal-cli will have to implement support for the storage service (#604)

**Comment by exquo** [comment]
I've tried again and it has worked! Before, I did not have the "master" android device connected to the internet. But now I'm remembering (from [this post](https://signal.org/blog/private-contact-discovery/)), that that's where the contacts list is (the device, not the internet :). 

Out of curiosity: how does signal-cli get the v2 groups info initially, on linking? Is there a different, non-"sync" message exchange then?
